app/routes/endemics/date-of-testing.js

```diff
@@ -189,15 +190,29 @@ module.exports = [
         }
       },
       handler: async (request, h) => {
-        const { dateOfVisit } = session.getEndemicsClaim(request)
+        const { dateOfVisit, typeOfReview, previousClaims } = session.getEndemicsClaim(request)
         const dateOfTesting = request.payload.whenTestingWasCarriedOut === 'whenTheVetVisitedTheFarmToCarryOutTheReview'
           ? dateOfVisit
           : new Date(
             request.payload['on-another-date-year'],
             request.payload['on-another-date-month'] - 1,
             request.payload['on-another-date-day']
           )
+
+        if (!isWithIn4MonthsBeforeOrAfterDateOfVisit(dateOfVisit, dateOfTesting) && typeOfReview === claimType.review) {
+          return h.view(endemicsDateOfTestingException, { backLink: pageUrl, ruralPaymentsAgency, exception: 'exceptionOne' }).code(400).takeover()
+        }
+        if (!isWithIn4MonthsBeforeOrAfterDateOfVisit(dateOfVisit, dateOfTesting) && typeOfReview === claimType.endemics) {
+          return h.view(endemicsDateOfTestingException, { backLink: pageUrl, ruralPaymentsAgency, exception: 'exceptionTwo' }).code(400).takeover()
```

Comment:
Can you rename 'exceptionOne', 'exceptionTwo', and 'exceptionThree' to something more meaningful?